cola/widgets/text.py

```diff
@@ -292,6 +291,28 @@ def wheelEvent(self, event):
             return
         return super(TextEdit, self).wheelEvent(event)
 
+    def should_override_tab_with_spaces(self, event):
+        return prefs.override_tab_with_spaces() and event.key() == Qt.Key_Tab
+
+    def keyPressEvent(self, event):
+        override_tab = self.should_override_tab_with_spaces(event)
+        if override_tab:
+            tab_width = self.ext.tabwidth()
+            new_press_event = QtGui.QKeyEvent(QtGui.QKeyEvent.KeyPress, Qt.Key_A, Qt.NoModifier, tab_width * ' ')
+            new_release_event = QtGui.QKeyEvent(QtGui.QKeyEvent.KeyRelease, Qt.Key_A, Qt.NoModifier, tab_width * ' ')
+            event.accept()
+            QtCore.QCoreApplication.postEvent(self, new_press_event)
+            QtCore.QCoreApplication.postEvent(self, new_release_event)
```

Comment:
Hmm I wonder if it'd be simpler to get the text cursor and insert the spaces instead of creating new events that insert spaces.

It's also kinda funky and probably not advised to send a "Key" press event that sends a non-key (aka multiple spaces).  I'm surprised that works.

Instead of posting new events, grab the textcursor and insert spaces.  That should be more robust.